sdk/src/main/java/io/opentelemetry/sdk/trace/export/MultiSpanExporter.java

```diff
@@ -56,6 +64,24 @@ public void shutdown() {
     }
   }
 
+  // Returns a merged error code, see the rules in the code.
+  private static ResultCode mergeResultCode(
+      ResultCode currentResultCode, ResultCode newResultCode) {
+    // If both errors are success then return success.
+    if (currentResultCode == SUCCESS && newResultCode == SUCCESS) {
+      return SUCCESS;
+    }
+
+    // If any of the codes is none retryable then return none_retryable;
+    if (currentResultCode == FAILED_NONE_RETRYABLE || newResultCode == FAILED_NONE_RETRYABLE) {
```

Comment:
If either one is FAILED_RETRYABLE the other one is FAILED_NONE_RETRYABLE, shouldn't it return FAILED_RETRYABLE since there actually are retryable spans/batches rather than none?

Reply:
We can make a case for both:
1. Yes there may be an exporter that can work if we retry.
1. We will never gate SUCCESS.

I think the first point is probably more important.

Reply:
If any of the exporters returns FAILED_NONE_RETRYABLE it means if we retry it it will failed again, which means we shouldn't retry, so we must return FAILED_NONE_RETRYABLE to our caller, otherwise our caller may retry the exact same data and it will fail again, so we will be stuck in infinite retries.

To support partial retrying when one of the exporters returns FAILED_NONE_RETRYABLE but others return FAILED_RETRYABLE we would need to to know which one's to retry next time and which one we shouldn't retry. That doesn't seem to be reasonably doable since we would not only need to remember which exporter returned what but we would also need to know if we are attempting to send the same data. 

What we have in this code I believe is the best that can be reasonably done.